### DISCUSSION TITLE
Sending more than one variable from a Shelly EM3 using a Rule and WebSend?

### DISCUSSION THREAD

**Discussion by hagenbuch** [arendst/Tasmota#17176] [Q&A]
With y'all's help I managed to flash Tasmota on a Shelly 3EM tri phase energy counter (over the air!) and also send the total energy counter to my webserver via a rule and WebSend:

    Rule1 ON tele-ENERGY#Total>0 DO WebSend [example.com] /?id=mydeviceid&time=%utctime%&uptime=%uptime%&bezug=%value% ENDON
    TelePeriod 60
    Rule1 1

However, while the console shows me the entire JSON object from the measurement

    SENSOR = {"Time":"2022-11-24T09:59:03","ENERGY":{"TotalStartTime":"2022-11-22T13:42:58","Total":-0.000,"Yesterday":-0.000,"Today":0.000,"ExportActive":[0.000,0.000,0.000],"Period":[ 0, 0, 0],"Power":[ 0,-0,-0],"ApparentPower":[ 3, 0, 0],"ReactivePower":[ 3, 0, 0],"Factor":[0.03,0.00,0.00],"Frequency":[50,57,171],"Voltage":[237, 0, 0],"Current":[0.013,0.013,0.013],"CurrentNeutral":0.011}}

(Yes it's an unused device so far)

I tried may ways to extract what I think is the counter for energy production (negative energy summation) ExportActive to a variable and then it as one more GET variable to the webserver.

No success so far. I am using Tasmota 12.1.1, had to compile Tasmota via the web compiler so the ADE7880 chip is included.

Pulling out the value via a subpart of the rule like

    ON ENERGY#ExportActive DO  var1 %ENERGY#ExportActive ENDON

did not work, also 

    ON ENERGY#ExportActive[1] DO  var1 %ENERGY#ExportActive[1] ENDON

did not. Would be glad if someone could point me to the right ideas..

Andreas

**Comment by barbudor** [comment]
The trick is to pickup each individual value in a `var` using a separated rule and then use those var in the latest rule
I think you got that already

Then your syntax for extracting the value is incorrect. You should use `%value%` to retreive the value from the trigger like:
```
ON ENERGY#ExportActive[1] DO  var1 %value% ENDON
```
Do that for each element you are interrested in
And for the last one you can then use the WebSend with each `%varN%` you need

Hope this help

**Comment by hagenbuch** [comment]
Ah great, many thanks! The index [0] seems to hold the sum of all three lines [1] [2] and [3] I guess! (I will investigate!)

**Reply by barbudor** [reply]
Interresting
Docs is just talking about index 1 and +, but nothing with 0
![image](https://user-images.githubusercontent.com/2996042/203772962-345cc703-1d59-4d78-828e-c80a06eac523.png)

Feel free to add something using the pencil button on the top right

**Reply by sfromis** [reply]
> The index [0] seems to hold the sum of all three lines [1] [2] and [3] I guess!

No. It seems to pick the first, like the "more correct" `[1]`

**Comment by hagenbuch** [comment]
Ah thanks. It would be great to have that sum, indices run from [0] to [3] anyway but maybe the N is being counted, too?

I will report a bug / suggestion anyway because apparently the "export" counters summarize (correctly) while the standard "total" counter runs backwards when energy is "exported". While there are counters in the market who show exactly that behaviour (old mechanical Ferraris counters) but any modern counter will summarize the "import" into one and the "export" into a different counter. Sold and bought electricity can have very different prices :-)
